docs/reference/attention_neural_bag_of_feature_learner.md

```diff
@@ -0,0 +1,339 @@
+This dataset has been preprocessed according to [[1]](#gru) and splitted into 5-fold cross validation protocol.  
+
+  ```python
+  from opendr.perception.heart_anomaly_detection.attention_neural_bag_of_feature.attention_neural_bag_of_feature_learner import \
+    AttentionNeuralBagOfFeatureLearner, get_cosine_lr_scheduler, get_AF_dataset 
+  ```
+
+  In order to download and construct the dataset, we will use the convenient function `get_AF_dataset` that has been imported above:
+
+  ```python
+  data_file = 'AF.dat'
+  fold_idx = 0
+  sample_length = 30
+  train_set, val_set, series_length, class_weight = get_AF_dataset(data_file, fold_idx, sample_length)
+  ```
+
+  In the above snippet, we specify the name of the data file, the index of the cross validation fold and the length of each sample (in seconds). Here, we don't specify any absolute path for the data file so it will be downloaded and saved under the name "AF.dat" in the current directory.   
+
+  Then we proceed to construct the learner object that will train the attention neural bag-of-feature network with 256 codewords and the temporal attention mechanism for 200 epochs starting from the learning rate of `2e-4` and gradually dropping to `1e-5` using a cosine scheduler. For the quantization layer, we will use the basic logistic version of the neural bag-of-features layer:  
+
+  ```python
+  learner = AttentionNeuralBagOfFeatureLearner(in_channels=1,
+                                               series_length=series_length,
+                                               n_class=4,
+                                               n_codeword=256,
+                                               quantization_type='nbof',
+                                               attention_type='temporal',
+                                               iters=200,
+                                               lr_scheduler=get_cosine_lr_scheduler(2e-4, 1e-5))
+  ```
+
+  After the learner has been constructed, we can train the learner using the `fit` function:
+   
+  ```python
+  performance = learner.fit(train_set, val_set, class_weight=class_weight) 
+  ```
+
+  In the above code, we pass the `class_weight` parameter to the `fit` function because the AF dataset is imbalanced. The `fit` function returns `performance`, which is a dictionary that contains the performance measured after each training epoch.   
+
+* **Download and evaluate pretrained model for the AF dataset**.  
+  In this example, we will show how pretrained models for the AF dataset [[2]](#af) can be easily downloaded by a single line of code using the functionality provided in `AttentionNeuralBagOfFeatureLearner` and evaluate the model on the test set. Since the AF dataset is divided into 5 folds, the following code iterates through each data split, downloads the corresponding pretrained model and evaluates on the correposnding validation set.  
```

Comment:
```suggestion
  In this example, we will show how pretrained models for the AF dataset [[2]](#af) can be easily downloaded by a single line of code using the functionality provided in `AttentionNeuralBagOfFeatureLearner` and evaluate the model on the test set.
  Since the AF dataset is divided into 5 folds, the following code iterates through each data split, downloads the corresponding pretrained model and evaluates on the corresponding validation set.  
```

Reply:
One sentence per line.